test/AppNotificationBuilderTests/APITests.cpp

```diff
@@ -382,5 +382,84 @@ namespace Test::AppNotification::Builder
             VERIFY_ARE_EQUAL(builder.BuildNotification().Payload(), expected);
         }
 
+        TEST_METHOD(AppNotificationAddProgressBar)
+        {
+            auto builder{ AppNotificationBuilder()
+                .AddText(L"Downloading this week's new music...")
+                .AddProgressBar(AppNotificationProgressBar()
+                    .BindTitle()
+                    .BindValueStringOverride()) };
+            auto expected{ L"<toast><visual><binding template='ToastGeneric'><text>Downloading this week's new music...</text><progress title='{progressTitle}' status='{progressStatus}' value='{progressValue}' valueStringOverride='{progressValueString}'/></binding></visual></toast>" };
+
+            VERIFY_ARE_EQUAL(builder.BuildNotification().Payload(), expected);
+        }
+
+        TEST_METHOD(AppNotificationAddMoreThanOneProgressBar)
+        {
+            auto builder{ AppNotificationBuilder()
+                .AddText(L"Downloading this week's new music...")
+                .AddProgressBar(AppNotificationProgressBar()
+                    .BindTitle()
+                    .BindValueStringOverride())
+                .AddProgressBar(AppNotificationProgressBar()
+                    .SetValue(0.8)
+                    .SetStatus(L"Still downloading..."))};
+            auto expected{ L"<toast><visual><binding template='ToastGeneric'><text>Downloading this week's new music...</text><progress title='{progressTitle}' status='{progressStatus}' value='{progressValue}' valueStringOverride='{progressValueString}'/><progress status='Still downloading...' value='0.8'/></binding></visual></toast>" };
+
+            VERIFY_ARE_EQUAL(builder.BuildNotification().Payload(), expected);
+        }
+
+        TEST_METHOD(AppNotificationProgressBarDefaults)
+        {
+            auto progressBar{ AppNotificationProgressBar() };
+            auto expected{ L"<progress status='{progressStatus}' value='{progressValue}'/>" };
+
+            VERIFY_ARE_EQUAL(progressBar.as<winrt::Windows::Foundation::IStringable>().ToString(), expected);
```

Comment:
I'm surprised this works since the object doesn't inherit IStringable in the idl, but maybe . I like the idea of processing everything through BuildNotification instead of casting to IStringable to simulate what devs would actually be doing.